pkg/util/helper/webstermethod.go

```diff
@@ -0,0 +1,159 @@
+func (pq *WebsterPriorityQueue) Less(i, j int) bool {
+	// In the Webster method, compare the priority of two Parties:
+	// the one with the higher value of Votes/(2*Seats+1) gets the next seat.
+	iPriority := float64(pq.Parties[i].Votes) / (float64(2*pq.Parties[i].Seats + 1))
+	jPriority := float64(pq.Parties[j].Votes) / (float64(2*pq.Parties[j].Seats + 1))
+	if iPriority == jPriority {
+		if pq.TieBreaker != nil {
+			return pq.TieBreaker(pq.Parties[i], pq.Parties[j])
+		}
+
+		// The party with more votes wins the tie.
+		if pq.Parties[i].Votes != pq.Parties[j].Votes {
+			return pq.Parties[i].Votes > pq.Parties[j].Votes
+		}
+
+		// The party with fewer seats wins the tie.
+		if pq.Parties[i].Seats != pq.Parties[j].Seats {
+			return pq.Parties[i].Seats < pq.Parties[j].Seats
+		}
+
+		// If both votes and seats are equal, the party with the lexicographically smaller name wins the tie.
+		// For example, party 'a' wins over party 'b' because 'a' < 'b' lexicographically.
+		return pq.Parties[i].Name < pq.Parties[j].Name
+	}
+
+	return iPriority > jPriority
+}
+
+// Swap swaps two Parties in the queue.
+func (pq *WebsterPriorityQueue) Swap(i, j int) {
+	pq.Parties[i], pq.Parties[j] = pq.Parties[j], pq.Parties[i]
+}
+
+// Push adds a new party to the queue.
+func (pq *WebsterPriorityQueue) Push(x interface{}) {
+	pq.Parties = append(pq.Parties, x.(Party))
+}
+
+// Pop removes and returns the party with the highest priority.
+func (pq *WebsterPriorityQueue) Pop() interface{} {
+	old := pq.Parties
+	n := len(old)
+	item := old[n-1]
+	pq.Parties = old[0 : n-1]
+	return item
+}
+
+// AllocateWebsterSeats allocates new seats using the Webster method.
+// newSeats: number of new seats to allocate across all Parties.
+// partyVotes: map of party name to number of votes.
+// initialAssignments: map of party name to initial seat assignments.
+// tieBreaker: function to break ties between two Parties.
+// Returns: slice of Party with updated Seats, sorted by Name in ascending order.
+func AllocateWebsterSeats(newSeats int32, partyVotes map[string]int64, initialAssignments map[string]int32, tieBreaker func(a, b Party) bool) []Party {
+	partiesMap := make(map[string]Party, len(partyVotes)+len(initialAssignments))
+
+	// set initial party assignments, the votes default to 0
+	// If a party is present in initialAssignments but not in partyVotes,
+	// it will be initialized with 0 votes and its initial seat count.
+	// Such a party will still be included in the allocation process, but
+	// the priority of such a party is 0, which is the lowest priority.
+	for n, s := range initialAssignments {
+		partiesMap[n] = Party{Name: n, Votes: 0, Seats: s}
+	}
+
+	for n, v := range partyVotes {
+		if p, ok := partiesMap[n]; ok {
+			p.Votes = v
+			partiesMap[n] = p
+		} else {
+			partiesMap[n] = Party{Name: n, Votes: v, Seats: 0}
+		}
+	}
+
+	pq := WebsterPriorityQueue{
+		Parties:    make([]Party, 0, len(initialAssignments)+len(partyVotes)),
```

Comment:
At this point, the length of partiesMap has already been determined. Would it be better to use `len(partiesMap)`?

Reply:
Good point!
Updated as suggested.